#732. Removing references from enrichment tags when removing judgments

madryk opened the pull request.

lukdumi commented on saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/reference/TagModifyingJudgmentReferenceRemover.java:
```diff
@@ -0,0 +1,128 @@
+        
+        List<EnrichmentTag> results = findTagsWithReferences(judgmentIds);
+        
+        if (!results.isEmpty()) {
```
In my opinion it is better to write 'if (results.isEmpty() {return;}' to get rid of nested curly braces.

madryk replied:
:+1:

lukdumi commented on saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/reference/TagModifyingJudgmentReferenceRemover.java:
```diff
@@ -0,0 +1,128 @@
+    @Override
+    public void removeReference(List<Long> judgmentIds) {
+        
+        List<EnrichmentTag> results = findTagsWithReferences(judgmentIds);
```
Why results? Why not just enrichmentTags?

madryk replied:
:+1:

lukdumi commented on saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/reference/TagModifyingJudgmentReferenceRemover.java:
```diff
@@ -0,0 +1,128 @@
+            enrichmentTagRepository.save(results);
+            
+            List<Long> judgmentsToReindex = results.stream().map(t -> t.getJudgmentId()).collect(Collectors.toList());
+            judgmentRepository.markAsNotindexed(judgmentsToReindex);
```
martAsNot<b>I</b>ndexed

madryk replied:
ok

lukdumi commented on saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/reference/TagModifyingJudgmentReferenceRemover.java:
```diff
@@ -0,0 +1,128 @@
+    }
+    
+    /**
+     * Builds query for selecting enrichment tags which reference to judgment need to be removed
```
... that refer to judgments that are subject to deletion...

madryk replied:
ok

lukdumi commented on saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/reference/TagModifyingJudgmentReferenceRemover.java:
```diff
@@ -0,0 +1,128 @@
+
+                removeReference(tagValue, judgmentIds);
+
+                String value = jsonStringWriter.write(tagValue);
```
why not jsonFormatter.format(tagValue)?

madryk replied:
because in package `pl.edu.icm.saos.common.json` already exists JsonFormatter and it does slightly different thing than `JsonStringWriter`

lukdumi commented on saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/reference/TagModifyingJudgmentReferenceRemover.java:
```diff
@@ -0,0 +1,128 @@
+    
+    protected final static String JUDGMENT_IDS_QUERY_PARAM_NAME = "judgmentIds";
+    
+    @Autowired
```
If you have setters then it's better to move @Autowired to them, the code is better to read then...

madryk replied:
ok

lukdumi commented on saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/reference/TagModifyingJudgmentReferenceRemover.java:
```diff
@@ -0,0 +1,128 @@
+    //------------------------ LOGIC --------------------------
+    
+    @Override
+    public void removeReference(List<Long> judgmentIds) {
```
removeReferences

madryk replied:
:+1:

lukdumi commented on saos-persistence/src/main/java/pl/edu/icm/saos/persistence/repository/IndexableObjectRepository.java:
```diff
@@ -16,4 +20,9 @@
+    @Modifying
+    @Transactional
+    @Query("update #{#entityName} i set i.indexed=false where i.indexed=true and i.id in :ids")
+    void markAsNotindexed(@Param("ids") List<Long> ids);
```
markAsNotIndexed

madryk replied:
:+1:

lukdumi commented on saos-persistence/src/main/java/pl/edu/icm/saos/persistence/repository/JudgmentRepositoryCustomImpl.java:
```diff
@@ -81,8 +78,6 @@ public void delete(List<Long> judgmentIds) {
         
         deleteJudgmentAttributesSql("judgment_lower_court_judgments", judgmentIds);
         
-        deleteEnrichmentTags(judgmentIds);
```
Why have you removed deletion of enrichment tags from this? What I meant yesterday what moving the deletion of enrichment tags to a separate repo/service and referring to it from here. If you delete judgment then you don't want to remember that there are two different services: one for removing the judgment alone and second for removing it along with corresponding tags, do you?

lukdumi commented on saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/delete/JudgmentWithEnrichmentDeleter.java:
```diff
@@ -0,0 +1,72 @@
+    
+    @Autowired
+    @Qualifier("judgmentReferenceRemover")
+    private JudgmentReferenceRemover judgmentReferenceRemover;
```
TagJudgmentReferenceRemover? Because now one can not know what reference it is about.

lukdumi commented on saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/reference/CompositeJudgmentReferenceRemover.java:
```diff
@@ -0,0 +1,38 @@
+    @Override
+    public void removeReference(List<Long> judgmentIds) {
+        
+        for (JudgmentReferenceRemover referringRemover : judgmentReferenceRemovers) {
```
referringRemover?

madryk replied:
changed to: `referenceRemover`

lukdumi commented on saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/delete/JudgmentWithEnrichmentDeleter.java:
```diff
@@ -0,0 +1,72 @@
+ * @author madryk
+ */
+@Service
+public class JudgmentWithEnrichmentDeleter {
```
I suggest to get rid of this class. See comments on JudgmentRepository.delete

lukdumi commented on saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/reference/TagModifyingJudgmentReferenceRemover.java:
```diff
@@ -0,0 +1,128 @@
+
+/**
+ * Class for removing reference to judgment from {@link EnrichmentTag}.
+ * It supports tags where reference removing is based on modifying {@link EnrichmentTag#getValue()}.
```
Abstract class for services that remove references to judgments from {@link EnrichmentTag}s.

madryk replied:
:+1:

lukdumi commented on saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/reference/PageableJudgmentReferenceRemover.java:
```diff
@@ -0,0 +1,56 @@
+ * 
+ * @author madryk
+ */
+public class PageableJudgmentReferenceRemover implements JudgmentReferenceRemover {
```
PageableDelegatingJudgmentReferenceRemover

madryk replied:
:+1:

lukdumi commented on saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/reference/PageableJudgmentReferenceRemover.java:
```diff
@@ -0,0 +1,56 @@
+ */
+public class PageableJudgmentReferenceRemover implements JudgmentReferenceRemover {
+    
+    private JudgmentReferenceRemover judgmentReferenceRemover;
```
delegatedJudgmentReferenceRemover

madryk replied:
:+1:

madryk commented:
`JudgmentWithEnrichmentDeleter` class was not deleted.
Added comments in `JudgmentWithEnrichmentDeleter` and updated comments in `JudgmentRepositoryCustom`

lukdumi commented:
We have to explain it. Leave it now as it is but create please an issue to solve it in the future.
